packages/blade/src/utils/styled-props/useStyledProps.web.tsx

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import type { CSSObject } from 'styled-components';
+import { getStyledProps } from './getStyledProps';
+import { getBaseBoxStyles, getDependencyProps } from '~components/Box/BaseBox/getBaseBoxStyles';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const useStyledProps = (props: Record<string, any>): CSSObject => {
+  const styledPropsStyles = getStyledProps(props);
+  const styledPropsMemoDependency = getDependencyProps(styledPropsStyles);
```

Comment:
Can you add some comments here to explain what these dependency props are and why we're memoizing it?

Reply:
added it to the new hook